server/chess.py

```diff
@@ -85,14 +96,10 @@ def get_moves_in_direction(
         while True:
             file += file_direction
             rank += rank_direction
-            target = self.get_piece(rank, file)
-            if target:
-                if target.side == piece.side:
-                    break
+            if validate_move_piece(self, piece, rank, file):
                 yield rank, file
```

Comment:
The only check that need be added here is that the rank and file are in range, this seems inefficient.